internal/opa/manager.go

```diff
@@ -0,0 +1,151 @@
+package opa
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	vspheremodel "github.com/kubev2v/forklift/pkg/controller/provider/model/vsphere"
+	"go.uber.org/zap"
+)
+
+var (
+	globalManager *Manager
+	globalMutex   sync.RWMutex
+)
+
+// Manager handles OPA initialization and VM validation coordination
+type Manager struct {
+	server  *Server
+	config  *Config
+	running bool
+	mutex   sync.RWMutex
+}
+
+// Config holds OPA configuration
+type Config struct {
+	Host           string
+	Port           string
+	PoliciesDir    string
+	StartupTimeout time.Duration
+}
+
+// NewManager creates a new OPA manager
+func NewManager() *Manager {
+	return &Manager{
+		config: &Config{
+			Host:           "127.0.0.1",
+			Port:           "8181",
+			PoliciesDir:    "./policies",
+			StartupTimeout: 30 * time.Second,
+		},
+	}
+}
+
+// Initialize sets up the OPA manager with policy discovery and server startup
+func (m *Manager) Initialize() error {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	// Discover policies directory
+	policiesDir := m.discoverPoliciesDirectory()
+	if policiesDir == "" {
+		return fmt.Errorf("no policies directory found")
+	}
+
+	m.config.PoliciesDir = policiesDir
+	zap.S().Named("opa").Infof("Using policies directory: %s", policiesDir)
+
+	// Create and initialize the OPA server
+	m.server = NewServer(m.config)
+	if err := m.server.Initialize(); err != nil {
+		return fmt.Errorf("failed to initialize OPA server: %w", err)
+	}
+
+	m.running = true
+	zap.S().Named("opa").Infof("OPA manager initialized successfully")
+	return nil
+}
+
+// IsRunning returns whether the OPA manager is currently running
+func (m *Manager) IsRunning() bool {
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+	return m.running
+}
+
+// ValidateVMs validates VMs using the OPA server
+func (m *Manager) ValidateVMs(vms *[]vspheremodel.VM) (*[]vspheremodel.VM, error) {
+	if !m.IsRunning() {
+		return vms, fmt.Errorf("OPA manager is not running")
+	}
+
+	return m.server.ValidateVMs(vms)
+}
+
+// Shutdown gracefully shuts down the OPA manager
+func (m *Manager) Shutdown() {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if m.server != nil {
+		m.server.Shutdown()
+	}
+	m.running = false
+	zap.S().Named("opa").Info("OPA manager shut down")
+}
+
+// discoverPoliciesDirectory searches for the policies directory in known locations
+func (m *Manager) discoverPoliciesDirectory() string {
+	// Check our two known policy locations
+	candidates := []string{
+		"./policies",    // Local development (make setup-opa-policies)
```

Comment:
I don't think this struct should assume that the policies are to be found here or there. It could received the policy dir as args in the constructor